### PR TITLE
My Home - remove "site created" banner and add to checklist.

### PR DESCRIPTION
**Pull request by getdave**
This PR works in conjunction with `code-D45830` to seek to address https://github.com/Automattic/wp-calypso/issues/42846 by:

* Removing the "Your site has been created" banner from the My Home page.
* Add the "Site Created" celebration as a checklist step which is automatically marked as completed.

Note this PR is responsible for the text of the new checklist task for "Site Created" only. All persistence is handled in `D45830-code`.

This PR also updates the  `<SiteSetupList>` component to only show "Action" buttons for the task if the text for these is provided in the task definition. This avoids having to show an action button for "Create your site" when this is not possible as the site is already created and it would make absolutely no sense whatsoever. 

## Screenshots

Annotation on 2020-07-02 at 11-18-50

## Testing instructions

* Apply D45830-code to your sandbox. 
* On sandbox change `if ( $site_created_date > strtotime( '2020-07-09' ) ) ` so that the date is _before_ today's date (for testing only!). Ensure these changes have been synced to your sandbox. See also `Testing with the PR` from instructions on D45830-code.
* Checkout this PR.
* Boot Calypso - `yarn start`. Wait for ages...
* Create a new site on WordPress.com via `http://calypso.localhost:3000/`.
* Once complete you should land on the "My Home" page.
* You should:
  - Not see the "Your site has been created banner" (see Issue).
  - See the `Site setup` checklist as the primary item at the top of the page.
  - See the first task `Site Created` selected and visible.
  - See the first task `Site Created` also marked as ✅ complete.
* You should also be able to click on the `Site Created` task step and view details about that task including a description.
* You should be able to click on the CTA "Get started" and advanced to the next incomplete task in the checklist.
* You should be able to reload the page and see the `Site Created` tasks remain ✅ and the first incomplete task selected.
* Add `?debug=true&view=VIEW_SITE_SETUP` to your URL to view the site setup view and verify the above still holds.
* Repeat the above, but this time instead of hitting the "Get started" CTA on the "Site Created" task, simply reload the page. You should see the "Site Created" task is complete, but not selected. Instead the next incomplete task should be selected. This confirms that user interaction is not required in order to mark the "Site Created" task as complete.
* Also verify that the "Site step" progress bar in the Calypso sidebar increments without a hard refresh as checklist tasks are updated. See https://github.com/Automattic/wp-calypso/pull/43861#issuecomment-655156660.
* Also verify that for a site registered _before_ 2020-07-09 that you do not see the "Site Created" step.

Addresses https://github.com/Automattic/wp-calypso/issues/42846

### CODE REVIEW

**Comment by getdave** [comment]
The following text which describes the checklist step for "Site Created" requires copy review:

> You created your site on WordPress.com. This is task was automatically completed when you created your new website.

Screen Shot 2020-07-02 at 10 30 56

**Comment by simison** [comment]
Would the original content in this step work? It would have quite a bit more delight and celebration. ;-)

image

**Comment by cpapazoglou** [comment]
I will agree with @simison. It makes sense to me to **land** to the `Site Created` tab with the following content
image

where `Get Started` will appear and actually move to the next step only if `Name your site` step is not completed. Same goes for `Next, we'll guide you through setting up and launching your site.`. This way we make sure that the user understands the process of going step by step.

**Comment by cpapazoglou** [comment]
Thanks for wrangling this @getdave !

I tested by test instructions and I can confirm that after creating a site, I land here


and the fist tab contains the following


I would suggest discussing about the content of the first tab and whether we land in the first or second tab as proposed above. Apart from that, LGTM!

**Comment by kristastevens** [comment]
Hello! Krista from Editorial here. Thanks for the ping! 

I really like the language @simison suggests above — it's celebratory and infuses the user with enthusiasm about continuing down the checklist. The tone is friendly as well and I think that's fitting here.

**Comment by getdave** [comment]
Thanks for review @cpapazoglou, @kristastevens and @simison.

> I would suggest discussing about the content of the first tab and whether we land in the first or second tab as proposed above. Apart from that, LGTM!

The pattern as it stands for the checklist is to select the first in-complete task in the list on component mount. Therefore to show the "Site Created" task as select even though it's complete, we'd need to add either:

* A one off conditional on the `site_created_ task - kinda feels hacky but gets the job done.
* Introduce a more abstract concept to represent the need to highlight a task even if it is complete - this could be `preferSelected` because it couldn't be exclusive of the possibility of any other tasks marking itself as being preferred for selection on load. The logic would need to select the first or last task which defines this property.

I'm not sure which option would be preferable or if another option is better. Perhaps @Automattic/serenity might know as (I believe) they built some of this.

-------

I can't say I'd be comfortable to merge this PR at this point. It needs the above fixed and edge cases testing. Also we need to approve the Dotcom diff.

**Comment by simison** [comment]
Can the task simply be unfinished and clicking on the button (or anything else) would mark it as done?

**Comment by getdave** [comment]
> Can the task simply be unfinished and clicking on the button (or anything else) would mark it as done?

I would be inclined to defer to @ianstewart on this one as the principal sponsor of this task.

My opinion is that it might be confusing to have a task which is patently "Done" to be marked as "Not Done" and then have to be manually checked off to continue. When you create a site the task "Site Creation" is automatically completed. Why should I as a user have to take additional action?

It would be great if we could come to a consensus here. Until that point, I'm not going to do any more work on the UI and focus on the broken Unit tests in the diff.

**Comment by mmtr** [comment]
> Can the task simply be unfinished and clicking on the button (or anything else) would mark it as done?

I like this too. The button can use the same "Get started" label we currently use. But I agree with @getdave above and having the task not marked as done can be confusing for users. I'd keep the task marked as complete with the "Get started" button moving users to the next task.

> A one off conditional on the `site_created_ task - kinda feels hacky but gets the job done.

I'd suggest to start with this. Early abstractions tend to be hard to iterate when we have only a one-off case, so I prefer to leave a more abstract implementation for later, once we note we have a similar pattern across different tasks.

**Comment by danhauk** [comment]
I definitely agree with others regarding the copy matching the original banner. The original intent of that was to provide a bit of celebration and delight to users since creating a site is a big milestone to celebrate.

I'd also +1 @getdave's thoughts and agree with not showing the task incomplete. The indicators in the list and the copy would be at odds with each other and cause some cognitive dissonance there (this text says my site is created, but where's my green checkmark?)

> When you create a site the task "Site Creation" is automatically completed. Why should I as a user have to take additional action?

I also agree with this but that's a trade-off I'm willing to compromise on, at least for this first implementation. We can iterate further on a more clever solution if we still don't see this moving the needle on the metrics we're looking for (checklist complete?)

So my suggestion would be, after creating the site the user is presented with the congratulatory messaging on the first task which is marked complete and `Get started` moves to the next task:

image

**Comment by getdave** [comment]
Thanks for all the feedback. Let's move forward with the MVP solution as suggested by @danhauk.

@danhauk once thing to note is that as it stands the code doesn't provide for having a _different_ title in the checklist sidebar and the main panel description. For example, this is what happens if I set the title to `Your site has been created!`:

Screenshot 2020-07-06 at 10 20 41

How do folks feel about launching with `Site created` in _both_ places and then I can follow up with another PR to add the ability to have different titles in the sidebar to realise the design you laid out above? I've keen to avoid a large PR here as it will become tricky to review.

**Comment by getdave** [comment]
Note to self:

What we're trying to do here is show a task item as both "selected" and "complete" but _only_ on the condition that the site has just been created. Otherwise, if the user has seen the "site created" step before then we just want to show as "completed" but not selected.

Tracking this "site created" view as having been seen is something that needs to be persisted else the next time the user returns to the My Home they will see the "Site Created" item in the checklist selected again.

**Comment by getdave** [comment]
I think this PR will be abandoned in favour of several smaller PRs. To that end I've raised an [initial PR to refactor the local state of `<SiteSetupList>` to make use of `useReducer`](https://github.com/Automattic/wp-calypso/pull/43911).

**Comment by gwwar** [comment]
The task needs to be added as a completed task (for newly created sites) from the checklist server library, please chat with us if this doesn't make sense

**Comment by getdave** [comment]
This PR now has a simple approach that works.

* User creates a site.
* User sees checklist with the following

Screen Shot 2020-07-06 at 16 41 50

* User clicks "Get started" and then sees...:

Screen Shot 2020-07-06 at 16 42 15

* If user reloads the My Home then `Site Created` will always be complete.

-------

Technically this is achieved by marking the `site_created` task as complete only once the user has seen the "Site Created" task in the UI for the first time. 

This allows us to initially show the "Site Created" task to the user. If we were to mark the task as "Complete" right away on the server side then the user would never see the task as the UI selects **the first _incomplete_ task** to show.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 1]
```diff
@@ -72,6 +72,22 @@ export const getTask = (
...
 	switch ( task.id ) {
+		case 'site_created':
+			taskData = {
+				timing: 1,
+				title: translate( 'Site created' ),
+				description: translate(
+					"Next, we'll guide you through setting up and launching your site."
+				),
+				actionText: 'Get started',
+				...( ! task.isCompleted && {
```
Comment (getdave) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:83]: Should never happen but just in case we make sure the action buttons complete the task.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 2]
```diff
@@ -72,6 +72,22 @@ export const getTask = (
...
+					actionDispatch: requestSiteChecklistTaskUpdate,
+					actionDispatchArgs: [ siteId, task.id ],
+				} ),
+				actionAdvanceToNext: true,
+				completeOnView: true,
```
Comment (getdave) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:88]: Indicates the task will be marked as complete once it has been view once.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 3]
```diff
@@ -72,6 +72,22 @@ export const getTask = (
...
+				title: translate( 'Site created' ),
+				description: translate(
+					"Next, we'll guide you through setting up and launching your site."
+				),
+				actionText: 'Get started',
+				...( ! task.isCompleted && {
+					actionDispatch: requestSiteChecklistTaskUpdate,
+					actionDispatchArgs: [ siteId, task.id ],
+				} ),
+				actionAdvanceToNext: true,
```
Comment (getdave) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:87]: Indicates the action advances the checklist to the next "incomplete" item.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 4]
```diff
@@ -72,6 +72,22 @@ export const getTask = (
...
+				timing: 1,
+				title: translate( 'Site created' ),
+				description: translate(
+					"Next, we'll guide you through setting up and launching your site."
+				),
+				actionText: 'Get started',
+				...( ! task.isCompleted && {
+					actionDispatch: requestSiteChecklistTaskUpdate,
+					actionDispatchArgs: [ siteId, task.id ],
+				} ),
```
Comment (getdave) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:86]: This is probably redundant and can be removed.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 5]
```diff
@@ -59,6 +59,10 @@ const startTask = ( dispatch, task, siteId ) => {
 	if ( task.actionDispatch ) {
 		dispatch( task.actionDispatch( ...task.actionDispatchArgs ) );
 	}
+
+	if ( task.actionAdvanceToNext ) {
+		advanceToNextInCompleteTask();
```
Comment (getdave) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:64]: Instead of passing a function we could pass the `localDispatch` from `<SiteSetupList>` and trigger the requisite action. 

Alternatively we could move all the selection state into Redux but that would be a larger PR and task.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 6]
```diff
@@ -125,6 +219,14 @@ const SiteSetupList = ( {
 		}
 	}, [ currentTaskId, dispatch, tasks ] );
 
+	// If specified then automatically complete the current task when viewed.
+	useEffect( () => {
+		if ( currentTask?.completeOnView ) {
+			setUserSelectedTask( true );
```
Comment (getdave) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:225]: Set to selected to ensure that it will be made (forced) visible even though it has been completed. This concept already exists with the component.

**Comment by obenland** [comment]
> Personally, neither option feels perfect, but the implementation in this PR as it stands at least sticks to the conventions and avoids forcing items to show when they are marked as "complete" in the state.

Agreed, option 1 feels better than 2

**Comment by jancavan** [comment]
> Ensure the "Site Created" task is auto-shown once only. Repeat views of the checklist should not re-show the "Site Complete" step.

I agree with this option, but to clarify, this will behave the way you mentioned?

> If the user reloads the My Home then Site Created will always remain complete and the next incomplete task will be selected.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 7]
```diff
@@ -4,7 +4,7 @@
 import { Card, Button } from '@automattic/components';
 import { isDesktop, isWithinBreakpoint, subscribeIsWithinBreakpoint } from '@automattic/viewport';
 import { translate } from 'i18n-calypso';
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useReducer } from 'react';
```
Comment (retrofox) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:7]: Please, confirm that `useState` is never used.

```suggestion
import React, { useEffect, useReducer } from 'react';
```

Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx]: Good spot. Fixed.

**Comment by simison** [comment]
Would it make sense to add an image to this step to bump up the celebratory nature little bit up? It feels little dry right now. Remember, they're landing here from signup, potentially after paying us a lot of money, and it's a major confirmation in their path. Sorry if I missed discussion about it, I tried skimming the convo a bit.

image

vs old one

image

**Comment by getdave** [comment]
> Would it make sense to add an image to this step to bump up the celebratory nature little bit up?

@simison Quite possibly. Seems like a good suggestion. Currently, I'm working to [this design](https://github.com/Automattic/wp-calypso/pull/43861#issuecomment-653317424) by @danhauk. If we want/need to add this then it should probably be in a followup PR.

If @danhauk agrees then I'm happy to raise a separate Issue.

**Comment by retrofox** [comment]
First tests looks :+1:

You should:
- [x] Not see the "Your site has been created banner" (see Issue).
- [x] See the Site setup checklist as the primary item at the top of the page.

Screen Shot 2020-07-07 at 7 36 57 AM 1

- [x] See the first task Site Created selected and visible.
- [x] See the first task Site Created also marked as ✅ complete.
- [x] You should also be able to click on the Site Created task step and view details about that task including a description.
- [x] You should be able to click on the CTA "Get started" and advanced to the next incomplete task in the checklist.
- [x] You should be able to reload the page and see the Site Created tasks remain ✅ and the first incomplete task selected.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js (hunk 4 above)
Comment (cpapazoglou) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:86]: Indeed, this task will always be completed and actions are not needed since you introduced `actionAdvanceToNext`

**Comment by getdave** [comment]
@retrofox Thanks for the 👍. I think I'll defer some of those items in order to keep the PR easy to review. Good areas to review though.

As discussed, I'd still like confidence check from @Automattic/serenity before this is merged.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 8]
```diff
@@ -125,6 +219,14 @@ const SiteSetupList = ( {
 		}
 	}, [ currentTaskId, dispatch, tasks ] );
 
+	// If specified then automatically complete the current task when viewed.
+	useEffect( () => {
+		if ( currentTask?.completeOnView ) {
```
Comment (cpapazoglou) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:224]: It seems that we can use `if ( currentTask?.completeOnView && ! currentTask.isCompleted ) {` instead of `if ( currentTask?.completeOnView ) {` so that we avoid running again the same code (and especially api calls) even for completed tasks. Tested locally and works, would like your opinion though.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx (hunk 6 above)
Comment (cpapazoglou) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:225]: I am not sure that `setUserSelectedTask` is appropriate here. It seems that it may be used for tracking user actions on tasks, in which case it would return wrong results since the user hasn't actually selected this task.
Also, I tried commenting it out and cannot see any visual difference (after creating a new site).

I may be missing something on the above though...

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 9]
```diff
@@ -94,6 +98,43 @@ const trackTaskDisplay = ( dispatch, task, siteId ) => {
 	);
 };
 
+const siteSetupListReducer = ( state, action ) => {
```
Comment (cpapazoglou) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:101]: Personally the way it was initially coded served me better. It seems that we add a lot of boilerplate code that doesn't enhance maintainability and also makes the code more error prone.

Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:101]: I agree with this. On 2nd thoughts I will revert the changes to local state back to how it was.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 10]
```diff
@@ -103,15 +144,68 @@ const SiteSetupList = ( {
...
+			currentDrillLayoutView: drillLayoutView,
+		} );
+
+	// Set isLoading
+	const isLoading = state.isLoading;
+	const setIsLoading = ( currentlyLoading ) =>
+		localDispatch( {
+			type: 'SET_IS_LOADING',
+			isLoading: currentlyLoading,
+		} );
```
Comment (gwwar) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:207]: Seems like a lot of boilerplate, was the previous component state getting hard to follow?

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 11]
```diff
@@ -248,17 +357,21 @@ const SiteSetupList = ( {
...
-								}
-							>
-								{ currentTask.actionText }
-							</Button>
+							{ currentTask.actionText && (
+								<Button
+									className="site-setup-list__task-action task__action"
+									primary
+									onClick={ () =>
+										startTask( dispatch, currentTask, siteId, advanceToNextIncompleteTask )
```
Comment (gwwar) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:365]: Non-blocking: If this is getting too long, we can change the startTask signature to an options object.

**Comment by gwwar** [comment]
Tested a number of scenarios and this looks pretty solid @getdave. I did note that the site setup list in the sidebar does not complete without a hard refresh. To reproduce complete the site setup list by skipping the site menu list.

I left a few other misc notes but nothing blocking.

Screen Shot 2020-07-07 at 2 42 18 PM

**Comment by gwwar** [comment]
Unrelated, but I found this in testing. Serenity missed this in the initial implementation but looks like the schema for `client/state/preferences/schema.js` needs to be relaxed to allow a boolean or object with the following shape, since we used the `dismissable-card` naming scheme in `wp-content/lib/home/cards.php#L144`.

@getdave let me know if you'd be interested in tidying that here or in a follow up. Basically this means we're throwing out indexedDB state for user preferences currently. This isn't harmful, but it defeats the purpose of trying to persist reducer state

Screen Shot 2020-07-07 at 2 30 20 PM

cc @kwight @mmtr

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx (hunk 8 above)
Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx]: Fixed.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx (hunk 6 above)
Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx]: I see why this is confusing. I think it's a naming problem.

>  It seems that it may be used for tracking user actions on tasks

It isn't...yet. But it could be. All it does is allow tasks to be "selected" (and viewed) by the User even if they are complete.

> Also, I tried commenting it out and cannot see any visual difference (after creating a new site).

Try commenting it out and then create a new site from scratch. You will land on the My Home page and the "Site created" item will be complete, but it will not be selected. Therefore you will miss the celebratory step of seeing the task content. 

Uncomment it and it will work as expected.

I think I'll rename the state and that will make the intent clearer.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx (hunk 11 above)
Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:274]: Noted.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js (hunk 4 above)
Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js:86]: However, if there was ever a situation where this was not automatically complete (API error...etc) then it would be good to retain the ability to complete the step manually. As this is added conditionally it should be ok to retain.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx (hunk 10 above)
Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx]: It was added so that I could pass the local reducer's `dispatch` into callbacks. However that requirement no longer exists so I've reverted all the changes to state back to how they were originally.

**Comment by getdave** [comment]
@retrofox Thanks for the review.

> LGTM. it works as expected.

Thanks! 👍 

> It's possible to simplify the code even more but probably something that we'd like to do after 🚢 . 

Turns out I decided to remove all the reducer changes as this actually _added_ complexity for no true benefit. There _was_ originally a good reason why I made those changes, but that reason no longer exists so it has no merit. Props to @cpapazoglou for this.

---------

@gwwar Thanks for the review.

> Tested a number of scenarios and this looks pretty solid @getdave. 

Good to hear.

>  I did note that the site setup list in the sidebar does not complete without a hard refresh. To reproduce complete the site setup list by skipping the site menu list.

I tested this and could not reproduce. I will add to testing instructions to ask others to verify.

**Comment by getdave** [comment]
> Unrelated, but I found this in testing. Serenity missed this in the initial implementation but looks like the schema for `client/state/preferences/schema.js` needs to be relaxed to allow a boolean or object with the following shape, since we used the `dismissable-card` naming scheme in `wp-content/lib/home/cards.php#L144`.
> 
> @getdave let me know if you'd be interested in tidying that here or in a follow up. Basically this means we're throwing out indexedDB state for user preferences currently. This isn't harmful, but it defeats the purpose of trying to persist reducer state
> 
> Screen Shot 2020-07-07 at 2 30 20 PM
> 
> cc @kwight @mmtr

I would be interested in learning about this and fixing it. If it's non-blocking I'd like to do it in a follow up PR to avoid blocking this change and introducing additional complexity. I'll raise an Issue.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx (hunk 9 above)
Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx]: Reverted.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 12]
```diff
@@ -125,6 +130,15 @@ const SiteSetupList = ( {
 		}
 	}, [ currentTaskId, dispatch, tasks ] );
 
+	// If specified, then automatically complete the current task when viewed
+	// if it is not already complete.
+	useEffect( () => {
+		if ( currentTask?.completeOnView && ! currentTask.isCompleted ) {
```
Comment (retrofox) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:136]: minor: should we check for `isCompleted` in the same way that `completeOnView`? Feel free to ignore it.

Comment (cpapazoglou) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:136]: Actually the api response contains `isCompleted` so I think it would be an overkill (whereas `completeOnView` is added later and may be `undefined`)

Comment (getdave) [reply client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx]: Harris is right but I've just done it anyway.

**Comment by getdave** [comment]
FYI I've renamed `site_created` to `start_site_setup` in response to @gwwar 's recommendation on the diff where she said:

> Hmm if we won't mark this as done [automatically], should we maybe name this something like start_site_setup instead or similar? A user's site is created at this point, but if folks click on the button we know they have intended to start setting it up.

This made perfect sense so I did a little renaming exercise. Update your diffs!

**Comment by getdave** [comment]
I will aim to merge this in the morning UTC.

**Comment by getdave** [comment]
This had to be rebased to resolve a conflict due to @cpapazoglou's PR being merged.
